### PR TITLE
Display auto-injection warning if the unsupported version of maven-pl…

### PR DESCRIPTION
**Pull request by c00ler**
…ugin is detected

A warning message is shown if an unsupported version of the `maven-plugin` is detected.

### CODE REVIEW

**Comment by c00ler** [comment]
Screenshot 2022-11-11 at 12 22 07

**Review by sorin-florea — APPROVED** [review]

**Review by welandaz — APPROVED** [review]

File: src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+                </f:entry>
+            </j:if>
 
-        <f:optionalBlock field="enabled" title="${%Enable auto-injection}" inline="true">
+            <j:if test="${instance.unsupportedMavenPluginWarningDetails != null}">
+                <f:entry>
+                    <div class="alert alert-danger">
+                        <l:icon class="icon-error icon-sm" alt="${%Unsupported Maven Integration plugin version detected}"/>
+                        Detected Maven Integration plugin version ${instance.unsupportedMavenPluginWarningDetails.mavenPluginVersion}.
+                        For auto-injection of the Gradle Enterprise Maven extensions, version ${instance.unsupportedMavenPluginWarningDetails.minimumSupportedVersion} is required.
```
Comment (guylabs) [inline src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly:20]: Extensions is plural.

**Review by guylabs — CHANGES_REQUESTED** [review]

File: src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java [hunk 2]
```diff
@@ -74,6 +75,16 @@ public boolean isShowLegacyConfigurationWarning() {
...
 
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public UnsupportedMavenPluginWarningDetails getUnsupportedMavenPluginWarningDetails() {
+        VersionNumber mavenPluginVersion = InjectionUtil.mavenPluginVersionNumber().orElse(null);
+
+        return (mavenPluginVersion != null && !InjectionUtil.isSupportedMavenPluginVersion(mavenPluginVersion))
+            ? new UnsupportedMavenPluginWarningDetails(mavenPluginVersion)
+            : null;
+    }
```
Comment (welandaz) [inline src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java:86]: Just a small nitpick: I find comparing on equality first a bit more easy to follow personally.
```
return mavenPluginVersion == null || InjectionUtil.isSupportedMavenPluginVersion(mavenPluginVersion)
        ? null
        : new UnsupportedMavenPluginWarningDetails(mavenPluginVersion);
```

File: src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly (hunk 1 above)
Comment (welandaz) [reply src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly:20]: It was plural is your comment [here](https://github.com/jenkinsci/gradle-plugin/pull/214#discussion_r1019660221), so I think @c00ler just posted it as is.

Comment (c00ler) [reply src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly:20]: Good catch! Yes, I just copied it. Will update.

File: src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java (hunk 2 above)
Comment (c00ler) [reply src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java:86]: Done.

File: src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly (hunk 1 above)
Comment (c00ler) [reply src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly]: Done

Comment (guylabs) [reply src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/config.jelly]: Right my bad. Can we use the following then on both places @c00ler ?

```
Detected Maven Integration plugin version x.x.x. For auto-injection of the Gradle Enterprise Maven extension, version x.x.x and above is required. Please upgrade the version of the Maven Integration plugin.
```

**Review by guylabs — APPROVED** [review]

**Pull request merged**